samples/java-valueentity-counter-spring-client/src/main/java/com/example/client/spring/rest/service/GrpcClientService.java

```diff
@@ -0,0 +1,40 @@
+package com.example.client.spring.rest.service;
+
+import com.example.CounterApi;
+import com.example.CounterServiceGrpc;
+import com.example.client.spring.rest.model.CounterRequest;
+import com.google.protobuf.Empty;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+// tag::decreaseCounterCall[]
+@Service
+public class GrpcClientService {
+
+  @Value("${as.host}")
+  String host;
+
+  @Value("${as.port}")
+  int port;
+
+  public String decrease(CounterRequest counterRequest) {
+    ManagedChannel channel = ManagedChannelBuilder.forAddress(host, port)
```

Comment:
I believe this should be created only once for the whole application and then we need to configure this Spring service to shutdown the channel when the application shutdowns
    
I mean, not per request, but init by the service constructor and shutdown when application stops.

Reply:
Makes sense, Updated it.